### ISSUE TITLE
System.Timers.Timer.TickImmediately

### ISSUE THREAD

**Issue by jez9999** [dotnet/runtime#54737]
## Background and Motivation

At the moment, `System.Timers.Timer` (which uses a `System.Threading.Timer` internally) has no way to set `dueTime`.  `dueTime` allows the caller to decide how long the initial delay before the first callback should be.  For some reason, `System.Timers.Timer` doesn't provide the option to configure this, and instead, always sets the value to the interval:

https://github.com/dotnet/runtime/blob/0416c3469b7bccff087b89db3d9967dfd36070a3/src/libraries/System.ComponentModel.TypeConverter/src/System/Timers/Timer.cs#L131

As users will frequently want to be able to trigger the timer immediately upon start, `dueTime` could be configurable on `System.Timers.Timer`; however, I propose a simpler `TickImmediately` bool on the class, which defaults to false (current behaviour), and can be set to true to set the underlying `System.Threading.Timer`'s `dueTime` to 0, giving an immediate tick upon starting the timer.

## Proposed API

Please provide the specific public API signature diff that you are proposing. For example:
```diff
namespace System.Timers
{
+    [TimersDescription(nameof(SR.TimerTickImmediately), null), DefaultValue(false)]
+    public bool TickImmediately
+    {
+        get;
+        set;
+    }
}
```

## Usage Examples

``` C#
var timer = new System.Timers.Timer();
timer.Interval = 60000; // Tick every minute
timer.Elapsed += async (sender, ea) => await doSomeWork(sender, ea);
timer.AutoReset = true; // Keep ticking automatically
timer.TickImmediately = true; // First tick is immediate
timer.Start();
```

## Alternative Designs

The `dueTime` integer could also be exposed directly.  This is a bit more complicated to implement, and I consider that it's not really necessary as generally the reason for this requirement is that the calling code will want the code to tick immediately, rather than after a different period than the interval.

## Risks

None as far as I can see.  Default behaviour of the Timer would remain the same.  Usage is clear and changed behaviour should be pretty obvious.

**Comment by AmrAlSayed0** [comment]
I think this would be better as a parameter to `.Start()` method since this an option that's only used at the start. No need to keep it around for the whole lifetime of the `Timer`, It would be like so:
```csharp
var timer = new System.Timers.Timer();
timer.Interval = 60000; // Tick every minute
timer.Elapsed += async (sender, ea) => await doSomeWork(sender, ea);
timer.AutoReset = true; // Keep ticking automatically
timer.Start(tickImmediately: true); // First tick is immediate
```
Also, another overload can be provided that takes a `TimeSpan` for the initial delay.
```csharp
var timer = new System.Timers.Timer();
timer.Interval = 60000; // Tick every minute
timer.Elapsed += async (sender, ea) => await doSomeWork(sender, ea);
timer.AutoReset = true; // Keep ticking automatically
var initialDelay = System.TimeSpan.FromSeconds(1);
timer.Start(initialDelay: initialDelay); // First tick is after 1 second
```

**Comment by jez9999** [comment]
@AmrAlSayed0 It would be appropriate for the lifetime of the timer if you wanted to stop and start the timer, immediately ticking each time.

**Comment by AmrAlSayed0** [comment]
@jez9999 Good point. I usually never restart a `Timer` so it never occured to me so I take back my mini proposal. 😅

**Comment by jez9999** [comment]
@AmrAlSayed0 So do you have any idea how to get these ideas reviewed?  This issue has been here a while without much comment.

**Comment by GSPP** [comment]
Alternatively, users could call the callback immediately.
